src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -250,231 +220,94 @@ public void AddPeer(ITxPoolPeer peer)
-            bool isReorg = (handlingOptions & TxHandlingOptions.Reorganisation) == TxHandlingOptions.Reorganisation;
-
-            if (_logger.IsTrace)
-                _logger.Trace(
-                    $"Adding transaction {tx.ToString("  ")} - managed nonce: {managedNonce} | persistent broadcast {isPersistentBroadcast}");
-
-            return FilterTransaction(tx, managedNonce, isReorg) ?? AddCore(tx, isPersistentBroadcast, isReorg);
-        }
-        
-        protected virtual AddTxResult? FilterTransaction(Transaction tx, in bool managedNonce, in bool isReorg)
-        {
-            Metrics.PendingTransactionsReceived++;
+            bool startBroadcast = (handlingOptions & TxHandlingOptions.PersistentBroadcast) == TxHandlingOptions.PersistentBroadcast;
 
-            if (tx.Hash is null)
-            {
-                Metrics.PendingTransactionsDiscarded++;
-                return AddTxResult.Invalid;
-            }
+            if (_logger.IsTrace) _logger.Trace(
+                $"Adding transaction {tx.ToString("  ")} - managed nonce: {managedNonce} | persistent broadcast {startBroadcast}");
 
-            if (!isReorg && _hashCache.Get(tx.Hash))
-            {
-                Metrics.PendingTransactionsKnown++;
-                return AddTxResult.AlreadyKnown;
-            }
-            
-            IReleaseSpec spec = _specProvider.GetSpec();
-            
-            if (!_validator.IsWellFormed(tx, spec))
-            {
-                // It may happen that other nodes send us transactions that were signed for another chain or don't have enough gas.
-                Metrics.PendingTransactionsDiscarded++;
-                if (_logger.IsTrace) _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, invalid transaction.");
-                return AddTxResult.Invalid;
-            }
-            
-            long gasLimit = Math.Min(BlockGasLimit ?? long.MaxValue, _txPoolConfig.GasLimit ?? long.MaxValue);
-            if (tx.GasLimit > gasLimit)
+            foreach (IIncomingTxFilter incomingTxFilter in _filterPipeline)
             {
-                Metrics.PendingTransactionsDiscarded++;
-                if (_logger.IsTrace) _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, gas limit exceeded.");
-                return AddTxResult.GasLimitExceeded;
-            }
-            
-            /* We have encountered multiple transactions that do not resolve sender address properly.
-             * We need to investigate what these txs are and why the sender address is resolved to null.
-             * Then we need to decide whether we really want to broadcast them.
-             */
-            if (tx.SenderAddress is null)
-            {
-                tx.SenderAddress = _ecdsa.RecoverAddress(tx);
-                if (tx.SenderAddress is null)
+                (bool accepted, AddTxResult? filteringResult) = incomingTxFilter.Accept(tx, handlingOptions);
+                if (!accepted)
```

Comment:
separate accepted seems redundant here

Reply:
consistent with the other filter, also limits the changes in the PR to make the code more familiar
later probably worth to reconsider
I gave it longer though of whether we should use string here, hcnage logging or maybe simplify return value. This was a the result of it. Not an ommission.